File: 백준/Silver/2161. 카드1/카드1.py
a = int(input())
card = []
for i in range(a):
  card.append(i + 1)

result = []
index = 0
while(True):
  result.append(card[0])
  del card[0]
  index += 1
  if index == a:
    break

  temp = card[0]
  for i in range(len(card) - 1):
    card[i] = card[i + 1]

  card[len(card) - 1] = temp
  
for item in result:
  print(item, end = ' ')

File: 백준/Silver/2161. 카드1/카드1.c
#include <stdio.h>

int main(void) {   
   int num, temp;
   scanf("%d", &num);

   int card[num];
   for(int i = 0; i < num; i++){
      card[i] = i + 1;
   }

   int index = 0;
   int result[num];
   while (1){
      if(card[0] == -1){
         break;
      }
      result[index] = card[0];
      index += 1;
      for(int i = 0; i < num - index; i++){
         card[i] = card[i + 1];
      }
      temp = card[0];
      
      for(int i = 0; i < num - index; i++){
         card[i] = card[i + 1];
      }
      card[num - index - 1] = temp;
      card[num - index] = -1;
   }

   for(int i = 0; i < num; i++){
      printf("%d ", result[i]);
   }
   
   return 0;
}